zk/txpool/pool_zk.go

```diff
@@ -259,6 +259,22 @@ func (p *TxPool) MarkForDiscardFromPendingBest(txHash common.Hash) {
 	}
 }
 
+// discards the transactions that are in overflowZkCoutners from pending
+// executes the discard function on them
+// deletes the tx from the sendersWithChangedState map
+// deletes the discarded txs from the overflowZkCounters
+func (p *TxPool) discardOverflowZkCountersFromPending(pending *PendingPool, discard func(*metaTx, DiscardReason), sendersWithChangedState map[uint64]struct{}) {
+	for _, mt := range p.overflowZkCounters {
+		log.Debug("[tx_pool] Removing TX from pending due to counter overflow", "tx", mt.Tx.IDHash)
```

Comment:
This one might be useful to have at info level.  It should be a rare thing that a single TX is enough to overflow an entire batch